File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.py
class Solution:
    def maxSubarrayLength(self, nums: List[int], k: int) -> int:
        him={}
        ans=float('-inf')
        l=0
        for r in range (len(nums)):
            him[nums[r]]=him.get(nums[r],0)+1
            while(him[nums[r]]>k):
                him[nums[l]]=him.get(nums[l],0)-1
                if(him[nums[l]]==0):
                    del him[nums[l]]
                l+=1
            ans=max(ans,r-l+1)
        return ans

File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.java
class Solution {
    public int maxSubarrayLength(int[] nums, int k) {
        Map<Integer,Integer> him=new HashMap<>();
        int ans=Integer.MIN_VALUE;
        int l=0;
        for(int i=0;i<nums.length;i++){
            him.put(nums[i],him.getOrDefault(nums[i],0)+1);
            while(him.get(nums[i])>k){
                him.put(nums[l],him.getOrDefault(nums[l],0)-1);
                if(him.get(nums[l])==0){
                    him.remove(nums[l]);
                }
                l++;
            }
            ans=Math.max(ans,i-l+1);
        }
        return ans;
    }
}